week2/Program.cs

```diff
@@ -0,0 +1,117 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+// Class excercises:
+app.MapGet("/excercise1", (string a, string b, string operation) =>
+{
+    var IsaValid = int.TryParse(a, out var parseda);
+    var IsbValid = int.TryParse(b, out var parsedb);
+    
+    if(!IsaValid || !IsbValid)
+    {
+        return "Invalid inputs";
+    }
+
+    if(operation == "add")
+    {
+        return $"The addition of {a} and {b} is {parseda+parsedb}";
+    }
+    else if (operation == "sub")
+    {
+        return $"The substraction of {a} and {b} is {parseda-parsedb}";
+    }
+    else if (operation == "multiply")
+    {
+        return $"The multiplication of {a} and {b} is {parseda*parsedb}";
+    } 
+    else{
+        return "Invalid Operator";
+    }
+});
+
+app.MapGet("/excercise2", (string input) =>
+{
+    var inputIsValid = int.TryParse(input, out var parsedInput);
+
+string AddNumbers(int parsedInput)
+{
+     var count=0;
+   
+    while (parsedInput > 0)
+    {
+     count = count + parsedInput % 10;
+     parsedInput = parsedInput / 10;
+    }
+    
+    return $"The count of {input} is {count}";
+}
+
+string CountCapitalLetters(string input)
+{
+    var count=0;
+
+    foreach (var character in input)
+    {
+        if(char.IsUpper(character))
+        {
+            count++;
+        }
+    }
+    return $"The number of capital letters in {input} is {count}";
+}
```

Comment:
These 2 methods dont necessarily have to be inside of the endpoint function. 
This kind of method (method inside of the method) is called the [local function](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions) and they cannot be used outside of the parent method. If you take them outside of the `app.MapGet("/excercise2", (string input) =>{...}` you can reuse them (if necessary) 
- We will maybe cover/mention this in our next class.